src/hawkbit-client.c

```diff
@@ -121,6 +121,34 @@ static size_t curl_write_to_file_cb(void *ptr, size_t size, size_t nmemb, struct
         return written;
 }
 
+/**
+ * @brief Add string to Curl headers, avoiding overwriting an existing
+ *        non-empty list on failure.
+ *
+ * @param[out] headers curl_slist** of already set headers
+ * @param[in]  string  Header to add
+ * @param[out] error   Error
+ * @return TRUE if string was added to headers successfully, FALSE otherwise (error set)
+ */
+static gboolean add_curl_header(struct curl_slist **headers, const char *string, GError **error)
+{
+        struct curl_slist *temp = NULL;
+
+        g_return_val_if_fail(string, FALSE);
```

Comment:
should also check for `curl_slist` being `NULL` as dereferencing below will fail otherwise.

Reply:
Done.